lib/intl.mjs

```diff
@@ -41,21 +43,60 @@ const descriptor = (value) => {
 const IntlDateTimeFormat = globalThis.Intl.DateTimeFormat;
 const ObjectAssign = Object.assign;
 
-export function DateTimeFormat(locale = IntlDateTimeFormat().resolvedOptions().locale, options = {}) {
+// Construction of  built-in Intl.DateTimeFormat objects is sloooooow,
+// so we'll only create those instances when we need them.
+// See https://bugs.chromium.org/p/v8/issues/detail?id=6528
+function getPropLazy(obj, prop) {
+  let val = obj[prop];
+  if (typeof val === 'function') {
+    val = new IntlDateTimeFormat(obj[LOCALE], val(obj[OPTIONS]));
+    obj[prop] = val;
+  }
+  return val;
+}
+// Similarly, lazy-init TimeZone instances.
+function getResolvedTimeZoneLazy(obj) {
+  let val = obj[TZ_RESOLVED];
+  if (typeof val === 'string') {
+    val = new TimeZone(val);
+    obj[TZ_RESOLVED] = val;
+  }
+  return val;
+}
+
+function deepClone(val) {
```

Comment:
Would it be possible to avoid the deep cloning altogether, and store `ro` instead of `options` for lazily creating new instances? (and use `ro.locale` as the locale?)

I ask because either deep- or shallow-cloning the object is going to be very tricky to do while matching the observable behaviour in [ToDateTimeOptions](https://tc39.es/ecma402/#sec-todatetimeoptions). But it seems like no matter whether we create a DateTimeFormat with the originally-given or the resolved options/locale, it should have the same result.

Reply:
@ryzokuken @sffc - @ptomato's idea above sounds like a good one. Will there be any difference in behavior between `i1` and `i2` below?

```js
i1 = new DateTimeFormat(locale, options);
ro = i1.resolvedOptions();
i2 = new DateTimeFormat(ro.locale, ro);
```

Reply:
Update: using `resolvedOptions()` output to replace `options` didn't work, because the polyfill adds additional options if the user's options don't contain unit options like `year` or `hour`. 

What *does* seem to work is a small variation of this approach: clone `resolvedOptions()`and delete all properties that weren't present in the original options object.

@ptomato - take a look at my latest commit and LMK if you think this is a promising approach.